cub/cub/device/dispatch/tuning/tuning_reduce.cuh

```diff
@@ -279,6 +372,94 @@ struct policy_hub
 
   using MaxPolicy = Policy1000;
 };
+
+struct arch_policies // equivalent to the policy_hub, holds policies for a bunch of CUDA architectures
+{
+  accum_type at;
+  op_type ot;
+  offset_size os;
+  int accum_size;
+
+  // IDEA(bgruber): instead of the constexpr function, we could also provide a map<int, reduce_arch_policy> and move the
+  // selection mechanism elsewhere
+
+  [[nodiscard]] _CCCL_API constexpr auto operator()(int arch) const -> reduce_arch_policy
+  {
+    if (arch >= 1000)
+    {
+      // use tuning if we have one, otherwise, pick it from arch 600
+      agent_reduce_policy rp{};
+      if (auto t = get_sm100_tuning(at, ot, os, classify_accum_size(accum_size)))
+      {
+        auto [scaled_items, scaled_threads] = scale_mem_bound(t->threads, t->items, accum_size);
+        rp                                  = agent_reduce_policy{
+          scaled_threads, scaled_items, t->items_per_vec_load, BLOCK_REDUCE_WARP_REDUCTIONS, LOAD_LDG};
+      }
+      else
+      {
+        rp = (*this)(600).reduce_policy;
+      }
+
+      const auto rp_nondet = agent_reduce_policy{
+        rp.block_threads,
+        rp.items_per_thread,
+        rp.vector_load_length,
+        BLOCK_REDUCE_WARP_REDUCTIONS_NONDETERMINISTIC,
+        rp.load_modifier};
+      return {rp, rp, rp, rp_nondet};
```

Comment:
Move this into the if above and then we can get rid of that ugly `else` branch. Doing the `(*this)(fallback)` dance feels icky, but as long as it can be avoided except in some really weird cases, it's fine. If it turns out we somehow need it more often, we should look for some more declarative abstraction to that I think.